features/step_definitions/basic_steps.rb

```diff
@@ -420,12 +420,23 @@ def assert_link_exists path, text
   find("#attendance_checkbox", visible: false).trigger('click')
 end
 
+
+When(/^I scroll to the bottom of the page$/) do
+  page.execute_script "window.scrollBy(0,10000)"
+end
+
 Then(/^I should( not)? see "([^"]*)" within "([^"]*)"$/) do |negated, project_title, project_list_area|
   within("##{project_list_area}") do
     if negated
       expect(page).to_not have_content(project_title)
     else
       expect(page).to have_content(project_title)
-    end  
+    end
+  end
+end
+
+Then(/^I should see "([^"]*)" within "([^"]*)":$/) do |project_title, project_list_area, table|
+  table.rows.each do |row|
+    step %Q{I should see "#{row[0]}" within "#{project_list_area}"}
```

Comment:
Don't call step definitions from other step definitions. To DRY this code, you can simply define a method within this file, e.g. right above your step definitions that use it.

Reply:
You don't make use of `project_title` here. Either say:
```ruby
 table.hashes.each do |row|
  title = row[project_title]
  # do sth. with it
end
```
Or even better hard code the `title` in the step definition. It is OK to have many step definitions.
```
Then I see the following projects in the list:
```

```ruby
 table.hashes.each do |row|
  title = row['Project Title']
  # do sth. with it
end
```